app/services/api/webflow-helpers.ts

```diff
@@ -25,61 +19,79 @@ export const cleanedSchedule = ({
   talks,
   recurringEvents,
 }: {
-  scheduledEvents?: RawScheduledEvent[]
-  speakers?: Speaker[]
-  workshops?: Workshop[]
-  talks?: Talk[]
-  recurringEvents?: RecurringEvents[]
-}): ScheduledEvent[] => {
+  scheduledEvents?: ScheduledeventsCollection[]
+  speakers?: CleanedSpeakers
+  workshops?: CleanedWorkshops
+  talks?: CleanedTalks
+  recurringEvents?: RecurringEvent[]
+}) => {
   return scheduledEvents
     ?.filter((schedule) => !schedule._archived && !schedule._draft)
-    .map((schedule) => ({
-      ...schedule,
-      location: WEBFLOW_MAP.location[schedule.location],
-      "recurring-event": recurringEvents?.find(({ _id }) => _id === schedule["recurring-event"]),
-      "speaker-2": speakers?.find(({ _id }) => _id === schedule["speaker-2"]),
-      day: WEBFLOW_MAP.scheduleDay[schedule.day] ?? WEBFLOW_MAP.scheduleDay["2e399bc3"],
-      talk: talks?.find((talk) => talk._id === schedule["talk-2"]),
-      type: WEBFLOW_MAP.scheduleType[schedule["event-type"]],
-      workshop: workshops?.find(({ _id }) => _id === schedule.workshop),
-    }))
+    .map((schedule) => {
+      return {
+        ...schedule,
+        location: WEBFLOW_MAP.location[schedule.location ?? ""] ?? "Unknown Location",
+        "recurring-event": recurringEvents?.find(({ _id }) => _id === schedule["recurring-event"]),
+        "speaker-2": speakers?.find((s) => s?._id === schedule["speaker-2"]),
+        "speaker-2-2": speakers?.find((s) => s?._id === schedule["speaker-2-2"]),
+        "speaker-3": speakers?.find((s) => s?._id === schedule["speaker-2-3"]),
+        day: (WEBFLOW_MAP.scheduleDay[schedule.day] ??
+          WEBFLOW_MAP.scheduleDay["2e399bc3"]) satisfies ScheduleDay,
+        talk: talks?.find((talk) => talk._id === schedule["talk-2"]),
+        type: WEBFLOW_MAP.scheduleType[schedule["event-type"] ?? ""] ?? "Unknown Type",
+        workshop: workshops?.find(({ _id }) => _id === schedule.workshop),
+      }
+    })
 }
 
+export type CleanedSchedule = ReturnType<typeof cleanedSchedule>
+
 /*
  * Converting speakers data from "type ids" to "type names"
  */
-export const cleanedSpeakers = (speakersData?: RawSpeaker[]): Speaker[] => {
+export const cleanedSpeakers = (speakersData?: SpeakersCollection[]) => {
   return speakersData?.map(cleanedSpeaker)
 }
 
-export const cleanedSpeaker = (speaker?: RawSpeaker): Speaker | null => {
+export type CleanedSpeakers = ReturnType<typeof cleanedSpeakers>
+
+export const cleanedSpeaker = (speaker?: SpeakersCollection) => {
   if (!speaker) return null
+
+  const speakerType: string | undefined = WEBFLOW_MAP.speakersType[speaker["speaker-type"]]
+  if (typeof speakerType === "undefined") {
+    return null
+  }
```

Comment:
Had trouble with inference here, not sure if this is how we want to handle this being null